src/course-unit/SubsectionUnitRedirect.test.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  initializeMocks, waitFor, render,
+} from '../testUtils';
+import SubsectionUnitRedirect from './SubsectionUnitRedirect';
+import { getXBlockApiUrl } from '../course-outline/data/api';
+
+let axiosMock;
+const courseId = '123';
+const subsectionId = 'block-v1+edX+DemoX+Demo_Course+type@sequential+block@19a30717eff543078a5d94ae9d6c18a5';
+const path = '/subsection/:subsectionId';
+const mockNavigate = jest.fn();
+const expectedCourseItemData = {
+  childInfo: {
+    children: [
+      {
+        id: 'unitId',
+      },
+    ],
+  },
+
+};
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockNavigate,
+}));
+
+const renderSubsectionRedirectPage = () => {
+  render(<SubsectionUnitRedirect courseId={courseId} />, {
+    path,
+    routerProps: {
+      initialEntries: [`/subsection/${subsectionId}`],
+    },
+  });
+};
+
+describe('subsection redirect page', () => {
+  beforeEach(async () => {
+    const mocks = initializeMocks();
+    axiosMock = mocks.axiosMock;
+    axiosMock
+      .onGet(getXBlockApiUrl(subsectionId))
+      .reply(200, expectedCourseItemData);
+  });
+
+  it('navigates to correct unit id', async () => {
```

Comment:
@imasdekar Can you kindly add a test case here for the subsection without any units as well?